test/with_dummyserver/test_proxy_poolmanager.py

```diff
@@ -651,6 +654,79 @@ def test_basic_ipv6_proxy(self) -> None:
 
 
 class TestHTTPSProxyVerification:
+    @staticmethod
+    def _get_proxy_fingerprint_md5(ca_path: str) -> str:
+        proxy_pem_path = pathlib.Path(ca_path).parent / "proxy.pem"
+        proxy_der = ssl.PEM_cert_to_DER_cert(proxy_pem_path.read_text())
+        proxy_hashed = hashlib.md5(proxy_der).digest()
+        fingerprint = binascii.hexlify(proxy_hashed).decode("ascii")
+        return fingerprint
+
+    def test_https_proxy_assert_fingerprint_md5(
+        self, no_san_proxy_with_server: Tuple[ServerConfig, ServerConfig]
+    ) -> None:
+        proxy, server = no_san_proxy_with_server
+        proxy_url = f"https://{proxy.host}:{proxy.port}"
+        destination_url = f"https://{server.host}:{server.port}"
+
+        proxy_fingerprint = self._get_proxy_fingerprint_md5(proxy.ca_certs)
+        with proxy_from_url(
+            proxy_url,
+            ca_certs=proxy.ca_certs,
+            proxy_assert_fingerprint=proxy_fingerprint,
+        ) as https:
+            https.request("GET", destination_url)
+
+    def test_https_proxy_assert_fingerprint_md5_non_matching(
+        self, no_san_proxy_with_server: Tuple[ServerConfig, ServerConfig]
+    ) -> None:
+        proxy, server = no_san_proxy_with_server
+        proxy_url = f"https://{proxy.host}:{proxy.port}"
+        destination_url = f"https://{server.host}:{server.port}"
+
+        proxy_fingerprint = self._get_proxy_fingerprint_md5(proxy.ca_certs)
+        new_char = "b" if proxy_fingerprint[5] == "a" else "a"
+        proxy_fingerprint = proxy_fingerprint[:5] + new_char + proxy_fingerprint[6:]
+
+        with proxy_from_url(
+            proxy_url,
+            ca_certs=proxy.ca_certs,
+            proxy_assert_fingerprint=proxy_fingerprint,
+        ) as https:
+            with pytest.raises(MaxRetryError) as e:
+                https.request("GET", destination_url)
+
+            assert "Fingerprints did not match" in str(e)
+
+    def test_https_proxy_assert_hostname(
```

Comment:
Once we add the `loopback_host` fixture to `san_proxy_with_server` we can change the `proxy_assert_hostname` to be `proxy.host`.